user/pages/05.blog/2018-01-09-docker-improvements/item.md

```diff
@@ -0,0 +1,171 @@
+In case you want to run the Java web application and the core services in separate containers, you can now define different health checks for each service.
+
+## Service Dependencies
+
+The old OpenNMS Horizon image had a sleep waiting for the PostgreSQL database.
+With Docker Compose version 2.1 and later it is possible to wait for a service condition.
+It makes the wait obsolete and OpenNMS Horizon is started when the PostgreSQL database is in state "healthy".
+
+## Java Process and PID 1
+
+Processes with ID 1 have special functionalities related to signal handling and cleaning up zombie processes.
+Using a current the example `docker-compose.yml` uses now the built-in `init: true` option, which runs an tiny init inside the container and prevents the Java process to get PID 1.
+The init option requires to use a Docker-Compose in version 2.2 or later.
+
+## Derive from the Horizon Image
+
+If you want to build and deploy your own image, just [fork](https://help.github.com/articles/fork-a-repo/) the [project](https://github.com/opennms-forge/docker-horizon-core-web).
+Create your custom Dockerfile for example named `MyDockerfile` with the following content:
+
+```
+FROM opennms/horizon-core-web:latest
+LABEL maintainer <its.me@acme.com>
+
+RUN yum -y install opennms-plugin-ticketer-jira \
+                   opennms-plugin-protocol-dhcp && \
+    yum clean all && \
+    rm -rf /var/cache/yum
+
+LABEL org.opennms.additional.packages="opennms-plugin-ticketer-jira, opennms-plugin-protocol-dhcp"
+```
+
+Build your image with
+
+```
+docker build -t my-opennms-horizon -f MyDockerfile .
+```
+
+An additional layer is created with the installed RPMS you provided in the `RUN` directive.
+Just as an optional idea, an additional label is created which tells you which packages you have installed.
+
+If you use as tag name `-t ${YOUR_DOCKERHUB_USER}/my-opennms-horizon:latest` you can push it to your DockerHub repository within
+
+```
+docker push ${YOUR_DOCKERHUB_USER}/my-opennms-horizon:latest
+```
+
+If you want to do this with CircleCI automatically you have to provide your DockerHub credentials as [project environment variable](https://circleci.com/docs/2.0/env-vars/#adding-environment-variables-in-the-app).
+The variables which are used in the config are `DOCKERHUB_LOGIN` and `DOCKERHUB_PASS`.
+
+## Container and Java Memory Management with Limits
+
+Running a Java application in a container has a few things to consider.
+To control and isolate resource usage of processes the Kernel feature _cgroups (Control Groups)_ is used.
+If you do nothing, the _Maximum Heap Size_ and _Initial Heap Size_ is calculated based on the memory of the Docker host system and can lead into problems when a memory limit on the container is set.
+Running a JVM on a server-class hardware means 1/64th of the machines physical memory is used as _Initial Heap Size_.
```

Comment:
...on server-class...
It's unclear what you mean by "server-class hardware," or how that impacts memory calculations. By context, it appears you mean "a machine with 64 GB of RAM."

Reply:
I've added a link what server-class in JVM means. Does it make sense now?